test/elm/external/external-test.ts

```diff
@@ -14,83 +14,83 @@ describe('Retrieve', () => {
     findRecordsSpy = sinon.spy(this.ctx, 'findRecords');
   });
 
-  it('should find conditions', function () {
-    const c = this.conditions.exec(this.ctx);
-    c.should.have.length(2);
+  it('should find conditions', async function () {
+    const c = await this.conditions.exec(this.ctx);
+    should(c).have.length(2);
```

Comment:
Just curious -- why the change from `c.should` to `should(c)`? I've just tested both ways and they seem to work the same.  Is this just a preference on your part or am I missing something?  Is that in case `c` comes back `undefined` for some reason?

Reply:
100% honesty: I don't fully know. This was a long time ago haha. I traced it back to https://github.com/projecttacoma/cql-execution/pull/18/commits/6e4fe50601059d88b71b804453fca2699dcc54ac

IIRC, I remember during the initial conversion, seeing a lot of tests failing due to "cannot read property "should" of undefined" which wasn't that helpful in tracing down the error. Wrapping it in the `should` function  call like this allowed for more useful messages like "expected <Promise> to have property `length`"? Short answer is that the tests gave better error messages when `c` would be null or undefined or something. Which I think helped trace down some errors in the conversion, but I'm not convinced it was actually necessary in all of these cases, oops

Reply:
OK.  No need to change it.  I was just curious.

Reply:
Something I just realized in addition to this is that using the `should` wrapper gives more type safety? For instance, in the old statement `c.should.have.length(2)`, `c` has type any, and so this relies on the developer to understand how you chain together `should` assertions. When using the `function`, the TypeScript LSP can do smart autocomplete:

<img width="706" alt="image" src="https://user-images.githubusercontent.com/16297930/193667201-3362c02b-bc28-4c61-b73f-f70796027507.png">

<img width="1197" alt="image" src="https://user-images.githubusercontent.com/16297930/193667301-6125da64-2c50-4709-a74d-95930b51c9b1.png">

Nice little boost to developer experience IMO

There might be a way to get the same thing with better configuration on using `should` with TypeScript, but not sure how that would work

Reply:
Oh, that is nice. I don't know if I noticed it _not_ working before -- but seeing it working makes me want it.